doc/spring-boot-plugin-dev.md

```diff
@@ -0,0 +1,318 @@
+```
+mvn clean package
+java -jar greeting-core/target/greeting-core-1.0-SNAPSHOT-exec.jar
+```
+
+The `greeting-core-1.0-SNAPSHOT-exec.jar` JAR does not contain any of the plugin code or Jackson dependency.
+When the application has started successfully, navigate to http://localhost:8080/greeting?name=Alice in your web browser to see the default greeting.
+
+Note that http://localhost:8080/greeting?name=Alice&language=fr does not display a multilingual greeting
+because the application is running without the language plugin.
+To run the application with the language plugin, run the following:
+
+```
+java -cp \
+	greeting-core/target/greeting-core-1.0-SNAPSHOT-exec.jar \
+	-Dloader.path=greeting-language-plugin/target/greeting-language-plugin-1.0-SNAPSHOT-jar-with-dependencies.jar \
+	org.springframework.boot.loader.PropertiesLauncher \
+	--greeting.config.external.location=classpath:/context.xml
+```
+
+Then, navigate to http://localhost:8080/greeting?name=Alice&language=fr to see a greeting in French.
+The plugin supports `en`, `fr`, `de` and `es` language codes (see the [JSON file](https://github.com/simonoakesepimorphics/spring-plugin-example/blob/main/greeting-language-plugin/src/main/resources/greeting.json) for details).
+
+In order to explain the development of this functionality, we must first describe the Maven project structure.
+
+## Project Structure
+
+This section describes how to organise and relate the Maven projects for the core application and its plugins.
+Each plugin is developed as a separate Maven project, which may be a module of the original project for convenience.
+Plugin development should not affect the packaging of the core application.
+
+Our example project contains the following Maven modules:
+
+* **spring-plugin-example** - Parent POM for all other modules.
+* **greeting-core** - Core web application containing the basic features and the plugin interface definition.
+    * Contains all Spring Boot and Spring WebMVC dependencies.
+* **greeting-language-plugin** - Plugin library containing the plugin implementation.
+    * Has **greeting-core** as a dependency with `provided` scope.
+
+Dependencies that the core application and its plugins have in common must be synchronised as much as possible,
+for example by sharing a common parent POM containing a `dependencyManagement` element which
+sets the versions for all shared dependencies.
+This is important for avoiding version conflicts in the deployed application.
+
+In each plugin POM, you must mark any dependencies in common with the core application (including the core application library itself, if it is present)
+with the `provided` scope (this makes them available for compilation but not included in the final JAR).
+The core application JAR will provide these dependencies at runtime, so there is no need to include them in the plugin JAR.
+The contents of these JARs are determined by the packaging process described in the next section.
+
+## Project Packaging
```

Comment:
This section is quite dry. It is all maven's fault. I don't think you can really make it more interesting because, well maven.